docs/guides/advanced/data/4844_data.md

```diff
@@ -0,0 +1,185 @@
+| Compute Polynomial Value:  | Calculate the value `p(x) = y`.                                                                                                                                                                                                              |
+| Prepare for Circuit Input: | Ensure that `hash(vh \|\| x \|\| y)` is provided as a [public input for the circuit][scheduler_expect_4844], based on the [Schwarz-Zippel lemma][schwarz_zippel] - if 2 polynomials are equal in a random point, there are equal everywhere. |
+
+### Inside the state keeper
+
+| step                         | explanation                                            |
+| ---------------------------- | ------------------------------------------------------ |
+| Collect data                 | Get `c`, `x`, `y`, `blob_proof` from witness generator |
+| Create New Transaction Type: | Form a new blob transaction with a special `0x3` type. |
+
+### Inside the L1 contract
+
+| step                             | explanation                                                                                                                                                                                                              |
+| -------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Receive Inputs:                  | Get `c`, `x`, `y`, and `blob_proof` as inputs from the [state keeper][executor_inputs].                                                                                                                                  |
+| Verification Hash:               | Obtain the verification hash through the BLOB_HASH precompile. [`_getBlobVersionedHash`][get_blob_vhash]                                                                                                                 |
+| Use Precompile for Verification: | Apply a [precompile][precompile_4844] to check if `p(x) == y` using the version hash, `c` and `blob_proof`. This is where the KZG scheme plays a crucial role, allowing the proof without needing the entire polynomial. |
+| Pass Public Input to Verifier    | Send `hash(vh \|\| x \|\| y)` as public input to the verifier using [`_batchAuxiliaryOutput`][executor_aux].                                                                                                             |
+
+This process simplifies how zkSync validates the attachment of 128kb of L2 data to L1, ensuring its correctness and
+integrity.
+
+### What, but why
+
+#### Why does prover need to compute the commitment
+
+Prover needs to come up with a 'random' point 'x' (otherwise Schwarz-Zippel lemma would not work), so it has to generate
+based on something. The simpler solution would be to generate it based on the hash of the pubdata, but unfortunately the
```

Comment:
as above, the point x is generates using the hash of the data...

Reply:
clarified